server/src/main/java/org/gluu/oxtrust/api/RegistrationManagementService.java

```diff
@@ -0,0 +1,210 @@
+        Response response = builder.build();
+        return response;
+    }
+
+    /**
+     * Saves the registration configuration data
+     *
+     * @param requestJson     request parameters
+     * @return response
+     */
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Saves configuration.",
+            notes = "Saves configuration.",
+            response = Response.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "invalid_request\n" +
+                    MALFORMED_REQUEST),
+            @ApiResponse(code = 403, message = "access_denied\n" +
+                    SERVER_DENIED_THE_REQUEST)
+
+    })
+    public Response saveConfiguration(String requestJson) {
+        try {
+            /* Note:
+             * Due to some properties of {@link GluuAttribute} class, direct JSON mapping may result in JSON exceptions.
+             * This may not be an issue on browser side, however, the java clients that are calling this method
+             * should not rely on automatic conversion of JSON objects to {@link RegistrationManagementResponse} class
+             * and should rather use {@link org.codehaus.jackson.map.DeserializationConfig.Feature} class <code>FAIL_ON_UNKNOWN_PROPERTIES</code> properties
+             * for deserialization.
+             */
+            RegistrationManagementRequest request = RegistrationManagementRequest.fromJson(requestJson);
+            final Boolean captchaDisabled = request.getCaptchaDisabled();
+            return save(captchaDisabled, request.getSelectedAttributes());
+        } catch (IOException jsonEx) {
+            log.error("Error while parsing request", jsonEx);
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        } catch (Exception otherEx) {
+            log.error(otherEx.getMessage(), otherEx);
+            return Response.serverError().build();
+        }
+    }
+
+    private Response save(boolean captchaDisabled, List<? extends GluuAttribute> selectedAttributes) {
+        Response.ResponseBuilder builder = Response.ok();
+        GluuOrganization org = organizationService.getOrganization();
+        RegistrationConfiguration config = org.getOxRegistrationConfiguration();
+        if (config == null) {
+            config = new RegistrationConfiguration();
+        }
+        config.setCaptchaDisabled(captchaDisabled);
+        List<String> attributeList = new ArrayList<String>();
+        for (GluuAttribute attribute : selectedAttributes) {
+            attributeList.add(attribute.getInum());
+        }
+        config.setAdditionalAttributes(attributeList);
+        org.setOxRegistrationConfiguration(config);
+        organizationService.updateOrganization(org);
+        jsonConfigurationService.saveOxTrustappConfiguration(getOxTrustAppConfiguration());
+        return builder.build();
+    }
+
+
+    private Response.ResponseBuilder search(String searchPattern) {
```

Comment:
All the logic is in this "presentation" class.
Also, we have duplicated code between the api and the old ui, now we'll have to maintain the code at two places.
Please refactor it, and take the logic out of this class.